setuptools/depends.py

```diff
@@ -51,7 +52,9 @@ def version_ok(self, version):
             and self.format(version) >= self.requested_version
         )
 
-    def get_version(self, paths=None, default: str = "unknown"):
+    def get_version(
+        self, paths=None, default: str | int = "unknown"
+    ) -> str | int | None:
```

Comment:
`get_version`, `get_module_constant`, and `extract_constant` *can* return `Any`, but I'm not sure that's within intended usage ?

Reply:
Yeah, this module is a bit annoying...

It looks very "vestigial", not very used in the setuptools code-base itself. But is exported as part of the API.

`get_module_constant` and `extract_constant` seem to be very likely returning something compatible with the return values of `ast.parse_literal` and possibly other objects that can be disassembled directly from Python bytecode. But I guess `Any` is the best thing we can do about that.

`get_version` is returning `None | Literal["default"] | T` if `self.format` is defined as `Callable[..., T]` or `Any` otherwise. But I don't think it is worth to add any form of validation here, we want to avoid breaking this API (potentially it is a good candidate for deprecation or moving to a different 3rd-party package).

Reply:
Ok so essentially `Any`, but I changed it to at least give some hints.